### PR TITLE
twister/samples: more filter cleanup for samples and a twister filter improvement

### PR DESCRIPTION
**Pull request by nashif**
- **twister: add depends_on_one key**
- **boards: add mmu/mpu as supported feature**
- **samples: userspace: improve filtering**
- **samples: mpu: refine filter and documentation**
- **samples: mpu: rename folder, this is not a test**

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

File: scripts/pylib/twister/twisterlib/config_parser.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ class TwisterConfigParser:
                        "min_ram": {"type": "int", "default": 8},
                        "modules": {"type": "list", "default": []},
                        "depends_on": {"type": "set"},
+                       "depends_on_one": {"type": "set"},
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/config_parser.py:61]: having its name as `depends_on_anyof` allows to add later `depends_on_oneof` (ie. exactly one of)

Comment (nashif) [reply scripts/pylib/twister/twisterlib/config_parser.py:61]: sure.

File: scripts/pylib/twister/twisterlib/testplan.py [hunk 2]
```diff
@@ -886,6 +886,11 @@ def apply_filters(self, **kwargs):
                     if dep_intersection != set(ts.depends_on):
                         instance.add_filter("No hardware support", Filters.PLATFORM)
 
+                if ts.depends_on_anyof:
+                    dep_intersection = ts.depends_on_anyof.intersection(set(plat.supported))
+                    if not dep_intersection:
+                        instance.add_filter("No hardware support", Filters.PLATFORM)
```
Comment (golowanow) [inline scripts/pylib/twister/twisterlib/testplan.py:892]: a bit different description from what 'depends_on' has, like
```suggestion
                        instance.add_filter("No hardware support (depends_on_anyof)", Filters.PLATFORM)
```

File: doc/develop/test/twister.rst [hunk 3]
```diff
@@ -391,6 +391,10 @@ depends_on: <list of features>
     A board or platform can announce what features it supports, this option
     will enable the test only those platforms that provide this feature.
```
Comment (golowanow) [inline doc/develop/test/twister.rst:392]:
```suggestion
    will enable the test only for those platforms that provide all these features.
```

**Review by golowanow — APPROVED** [review]

**Review by teburd — APPROVED** [review]

File: boards/snps/nsim/arc_classic/nsim_nsim_em.yaml [hunk 4]
```diff
@@ -8,6 +8,8 @@ toolchain:
   - zephyr
   - cross-compile
   - arcmwdt
+supported:
+  - mpu
```
Comment (evgeniy-paltsev) [inline boards/snps/nsim/arc_classic/nsim_nsim_em.yaml:12]: Hmm, I'm wondering why this the only SNPS platform we add mpu tag to?
There are a lot of snps platforms which have mpu but now they will be filtered out in mpu-related tests.

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**